src/main/scala/io/dronekit/CouchbaseStreamsWrapper.scala

```diff
@@ -292,6 +295,24 @@ class CouchbaseStreamsWrapper(host: String, bucketName: String, password: String
     observableToSource(docObservable).map(json => convertToEntity[T](json))
   }
 
+  /**
+   * Does a batch update of keys -> each update
+   * @param entities Seqence of BatchUpdateRequest to update
+   * @return a source of entities from the batch update request
+   */
+  def batchUpdate[T](entities: Seq[BatchUpdateRequest[T]])(implicit format: JsonFormat[T]): Source[DocumentResponse[T], Any] = {
+    val docObservable = Observable.from(entities.map{e => e.key})
+      .flatMap(key => entities.find(e => e.key == key) match {
+        case None => throw new DocumentNotFound(s"Could not find doc for key ${key}")
```

Comment:
Actually based off of this graphic it seems like throwing an error would cause all future items to also not be updated.. http://reactivex.io/documentation/observable.html